### DISCUSSION TITLE
Check if websocket is closed

### DISCUSSION THREAD

**Discussion by alxhotel** [uNetworking/uWebSockets.js#269] [Q&A] [answered]
If you call `ws.send` when the websocket is closed, it throws an error.

How can you check if the websocket has been closed before sending the message?

**Comment by ghost** [comment, marked as answer]
You can't. You have to track that.

**Comment by alxhotel** [comment]
Ok, thanks.

**Comment by alxhotel** [comment]
So, if the websocket is closed, is it guaranteed that the `onClose` event will be called before any `send` call?
